test/e2e/test_helpers.go

```diff
@@ -641,19 +641,30 @@ func (t testHelper) verifyCategoryExists(ctx context.Context, categoryKey, categ
 	Expect(err).ShouldNot(HaveOccurred())
 }
 
-func (t testHelper) verifyCategoriesNutanixMachines(ctx context.Context, clusterName, namespace string, expectedCategories map[string]string) {
-	nutanixMachines := t.getMachinesForCluster(ctx, clusterName, namespace, bootstrapClusterProxy)
-	for _, m := range nutanixMachines.Items {
-		machineProviderID := m.Spec.ProviderID
-		Expect(machineProviderID).NotTo(BeNil())
-		machineVmUUID := t.stripNutanixIDFromProviderID(*machineProviderID)
-		vm, err := t.nutanixClient.V3.GetVM(ctx, machineVmUUID)
-		Expect(err).ShouldNot(HaveOccurred())
-		categoriesMeta := vm.Metadata.Categories
-		for k, v := range expectedCategories {
-			Expect(categoriesMeta).To(HaveKeyWithValue(k, v))
-		}
-	}
+func (t testHelper) verifyCategoriesNutanixMachines(ctx context.Context, clusterName, namespace string, expectedCategories map[string][]string) {
+	Eventually(
+		func(g Gomega) {
+			nutanixMachines := t.getMachinesForCluster(ctx, clusterName, namespace, bootstrapClusterProxy)
+			for _, m := range nutanixMachines.Items {
+				machineProviderID := m.Spec.ProviderID
+				g.Expect(machineProviderID).NotTo(BeNil())
+				machineVmUUID := t.stripNutanixIDFromProviderID(*machineProviderID)
+				vm, err := t.nutanixClient.V3.GetVM(ctx, machineVmUUID)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				categoriesMeta := vm.Metadata.CategoriesMapping
```

Comment:
nit: this is not categories metadata but categories mapping. consider renaming the variable to indicate what it actually contains.